megamek/src/megamek/client/ui/swing/MovementDisplay.java

```diff
@@ -3983,31 +4038,63 @@ private TreeMap<Integer, Vector<Integer>> getDroppedUnits() {
         Set<Integer> alreadyDropped = cmd.getDroppedUnits();
         // cycle through the bays
         int bayNum = 1;
-        Vector<Bay> Bays = ce.getTransportBays();
-        for (int i = 0; i < Bays.size(); i++) {
-            Bay currentBay = Bays.elementAt(i);
+        Vector<Transporter> transporters = ce.getTransports();
+        List<Entity> currentUnits;
+
+        for (int i = 0; i < transporters.size(); i++) {
+            Transporter currentT = transporters.elementAt(i);
+            boolean isInfantryT = false;
+            currentUnits = new ArrayList<>();
             Vector<Integer> bayChoices = new Vector<>();
-            int doorsEligibleForDrop = currentBay.getCurrentDoors();
-            List<Entity> currentUnits = new ArrayList<>();
-            for (Entity entity : currentBay.getDroppableUnits()) {
-                if (alreadyDropped.contains(entity.getId())) {
-                    doorsEligibleForDrop--; //If a unit is set to drop from a bay, we should reduce our capacity
-                } else {
-                    currentUnits.add(entity);
+            int doorsEligibleForDrop = 0;
+
+            // Handle infantry first
+            if (currentT instanceof InfantryTransporter iTransporter) {
+                isInfantryT = true;
+
+                // Can't drop infantry from above 8 Altitude
+                if (isInfantryT) {
```

Comment:
I fail to understand this if clause

Reply:
Skips the current Transporter if it's an InfantryTransporter and the current / last altitude in the MP is > Altitude 8.

Reply:
isInfantryT is always true in l.4056 because l.4053 sets it to true.

Reply:
Aha, not anymore!
But also, there was a lot of refactoring going on >_>